glean-core/uniffi/src/glean.udl

```diff
@@ -4,14 +4,25 @@ namespace glean {
 
     void glean_enable_logging();
 
-    void glean_set_upload_enabled(boolean enabled);
+    void glean_set_upload_enabled(boolean enabled, OnUploadEnabledChanges changes_callback);
 
     // Experiment reporting API
     void glean_set_experiment_active(string experiment_id, string branch, record<DOMString, string> extra);
     void glean_set_experiment_inactive(string experiment_id);
     RecordedExperiment? glean_test_get_experiment_data(string experiment_id);
 };
 
+callback interface OnUploadEnabledChanges {
+    // Called when upload is to be disabled
+    void will_be_disabled();
+
+    // Called when upload was toggled from off to on.
+    void is_enabled();
+
+    // Called when upload was toggled from on to off.
+    void is_disabled();
+};
```

Comment:
@travis79 
What do you think about the naming? Is it self-explanatory enough when each of them will be called?

Reply:
I think they are fine, the only thing I could think of as an alternative is to use `on` as a prefix rather than `is`. `onEnabled` and `onDisabled` sound more like callback functions where `is` tends to make me think of them as property accessors.